### PR TITLE
feat(resource-detector-azure): Add AKS Resource Detector

### PR DESCRIPTION
**Pull request by JacksonWeber**
## Short description of the changes
This pull request adds support for detecting Azure Kubernetes Service (AKS) environments in the `@opentelemetry/resource-detector-azure` package. It introduces a new resource detector for AKS clusters, enabling automatic extraction of cluster metadata such as cluster name and resource ID, and updates documentation and semantic conventions accordingly. The changes also include comprehensive tests for the new detector.

**AKS Resource Detection:**

* Added `AzureAksDetector` in `src/detectors/AzureAksDetector.ts`, which detects AKS cluster metadata from the environment or a mounted ConfigMap file, and exposes the attributes `cloud.provider`, `cloud.platform`, `cloud.resource_id`, and `k8s.cluster.name`.
* Exported the new `azureAksDetector` from `src/detectors/index.ts` and `src/index.ts` for public use. [[1]](diffhunk://#diff-6c7e6ee102d16fb9411421fd04e912114756748cca90a1397fbaff6c93ce88ceR17) [[2]](diffhunk://#diff-630bf05b1cca6e0562642eae9a2675d984993e43e3e6fe93326d54f2f519fee1R18)

**Semantic Convention Updates:**

* Added the experimental attribute `k8s.cluster.name` and enum value `azure.aks` for `cloud.platform` in `src/semconv.ts`. [[1]](diffhunk://#diff-54e876d474ba76bfd2e29aec58a84c3db5a41236533a139664f3c5037a6c31a3R70-R78) [[2]](diffhunk://#diff-54e876d474ba76bfd2e29aec58a84c3db5a41236533a139664f3c5037a6c31a3R211-R219)
* Defined constants and helper functions for AKS metadata extraction in `src/types.ts`, including parsing cluster name from ARM resource IDs.

**Documentation and Testing:**

* Updated `README.md` with AKS detector usage instructions, attribute descriptions, and configuration examples.
* Added unit tests for AKS detection logic in `test/detectors/AzureAksDetector.test.ts`, covering environment variable and file-based detection, and edge cases.

### CODE REVIEW

File: packages/resource-detector-azure/src/types.ts [hunk 1]
```diff
@@ -36,6 +36,44 @@ export const AZURE_VM_METADATA_PATH =
...
+  resourceId?: string;
+}
+
+/**
+ * Extracts the cluster name from an AKS ARM resource ID.
+ * @param resourceId The full ARM resource ID, e.g.:
+ *   /subscriptions/.../resourceGroups/.../providers/Microsoft.ContainerService/managedClusters/my-cluster
+ * @returns The cluster name (last segment), or undefined if not found
+ */
+export function extractClusterNameFromResourceId(
```
Comment (hectorhdzg) [inline packages/resource-detector-azure/src/types.ts:58]: Looks this method is in wrong file, as is not a type
